src/main/java/com/microsoft/jenkins/containeragents/util/DockerRegistryUtils.java

```diff
@@ -0,0 +1,35 @@
+package com.microsoft.jenkins.containeragents.util;
+
+
+import org.apache.commons.lang.StringUtils;
+
+public final class DockerRegistryUtils {
+
+    public static String formatUrlToWithProtocol(String url) {
+        if (StringUtils.isBlank(url)) {
+            return url;
+        }
+        if (url.toLowerCase().startsWith("http://") || url.toLowerCase().startsWith("https://")) {
+            return url;
+        } else {
+            return "https://".concat(url);
+        }
+    }
+
+    public static String formatUrlToWithoutProtocol(String url) {
+        if (StringUtils.isBlank(url)) {
+            return url;
+        }
+        if (url.toLowerCase().startsWith("http://")) {
+            return StringUtils.substringAfter(url.toLowerCase(), "http://");
```

Comment:
Technically URL is case sensitive, so better to use string.substring("http://".length)